profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+      $url = NewsroomHelper::buildImportUrl(NEXTEUROPA_NEWSROOM_ITEM_FULL);
+    }
+
+    return $url;
+  }
+
+  /**
+   * Builds import url.
+   *
+   * @param string $url_part
+   *   Part of URL.
+   *
+   * @return string
+   *   URL.
+   */
+  public static function buildImportUrl($url_part) {
+    global $_newsroom_universe_url;
+    return $_newsroom_universe_url . $url_part;
+  }
+
+  /**
+   * Get newsroom item by Id.
+   */
+  public static function getNodeByNewsroomItemId($newsroom_id = 0) {
+    $newsroom_data = &drupal_static(__FUNCTION__);
+    $cache_key = 'newsroom_item_by_id:' . $newsroom_id;
+    if (!isset($newsroom_data)) {
+      $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+      if ($cache) {
+        $newsroom_data = $cache->data;
+      }
+      else {
+        $query = new EntityFieldQuery();
+        $newsroom_item = $query
+          ->entityCondition('entity_type', 'node')
+          ->fieldCondition('field_newsroom_item_id', 'value', $newsroom_id, '=')
+          ->execute();
+        $newsroom_data = isset($newsroom_item['node']) ? $newsroom_item['node'] : FALSE;
+        cache_set($cache_key, $newsroom_data, NEXTEUROPA_NEWSROOM_CACHE_TABLE, time() + 3600);
+      }
+    }
+    return $newsroom_data;
+  }
+
+  /**
+   * Finds a menu item for a given path.
+   */
+  public static function getMenuByPath($path) {
+    $cache_key = 'get_menu_by_path' . md5($path);
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $data = $cache->data;
+    }
+    else {
+      $query = db_select('menu_links', 'ml')
+        ->fields('ml')
+        ->condition('ml.link_path', $path)
+        ->execute();
+      $data = NewsroomHelper::convertPdoToArray($query);
+      cache_set($cache_key, $data, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+    return $data;
+  }
+
+  /**
+   * Helper to retrieve a mlid for a given path.
+   *
+   * @param string $topic_ids
+   *   Topic term ids.
+   *
+   * @return array
+   *   Field data.
+   */
+  public static function getNodesAssociatedToTopics($topic_ids) {
+    $topic_ids = is_array($topic_ids) ? $topic_ids : array($topic_ids);
+    $cache_key = 'get_topic_entities' . md5(implode('', $topic_ids));
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $data = $cache->data;
+    }
+    else {
+      $query = db_select('field_data_field_associated_newsroom_topic', 'f')
+        ->fields('f')
+        ->condition('f.field_associated_newsroom_topic_tid', $topic_ids)
+        ->execute();
+      $data = NewsroomHelper::convertPdoToArray($query);
+
+      cache_set($cache_key, $data, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+    return $data;
+  }
+
+  /**
+   * Get combined node newsroom topics.
+   *
+   * @param object $node_wrapper
+   *   Node wrapper.
+   *
+   * @return array
+   *   Newsroom topics.
+   */
+  public static function getNodeTopics($node_wrapper) {
```

Comment:
hopefully this module won't handle entity_metadata_wrapper generated object but the node object itself, as i reported in .module parts (not in github, but in wiki).